src/main/java/org/opensearch/knn/index/KNNIndexShard.java

```diff
@@ -122,7 +122,8 @@ public void warmup() throws IOException {
 
             for (FieldInfo fieldInfo : reader.getFieldInfos()) {
                 if (fieldInfo.attributes().containsKey(KNNVectorFieldMapper.KNN_FIELD)) {
-                    SpaceType spaceType = SpaceType.getSpace(fieldInfo.attributes().get(SPACE_TYPE));
+                    String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, SpaceType.L2.getValue());
```

Comment:
How about to add a default value here to make more readable?
```suggestion
                    String spaceTypeDefaultVaule = SpaceType.L2.getValue();
                    String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, spaceTypeDefaultVaule);
```

Reply:
I would make default value a constant

Reply:
That makes sense, I will update.

Reply:
Either way is fine, but i am leaning towards what you have :) since method itself gives the hint that it is get or Default :)

Reply:
Right, I think I will just keep what I have. So, we do have SpaceType.DEFAULT, but in this case I do not want to use that because I know for sure that this should be L2. I will update with a comment.